internal/datastore/spanner/spanner.go

```diff
@@ -239,18 +240,50 @@ func (sd *spannerDatastore) SnapshotReader(revisionRaw datastore.Revision) datas
 	return spannerReader{executor, txSource, sd.filterMaximumIDCount}
 }
 
+func (sd *spannerDatastore) readTransactionMetadata(ctx context.Context, transactionTag string) (map[string]any, error) {
+	row, err := sd.client.Single().ReadRow(ctx, tableTransactionMetadata, spanner.Key{transactionTag}, []string{colMetadata})
+	if err != nil {
+		if spanner.ErrCode(err) == codes.NotFound {
+			return map[string]any{}, nil
+		}
+
+		return nil, err
+	}
+
+	var metadata map[string]any
+	if err := row.Columns(&metadata); err != nil {
+		return nil, err
+	}
+
+	return metadata, nil
+}
+
 func (sd *spannerDatastore) ReadWriteTx(ctx context.Context, fn datastore.TxUserFunc, opts ...options.RWTOptionsOption) (datastore.Revision, error) {
 	config := options.NewRWTOptionsWithOptions(opts...)
 
 	ctx, span := tracer.Start(ctx, "ReadWriteTx")
 	defer span.End()
 
+	transactionTag := "sdb-rwt-" + uuid.NewString()
+
 	ctx, cancel := context.WithCancel(ctx)
-	ts, err := sd.client.ReadWriteTransaction(ctx, func(ctx context.Context, spannerRWT *spanner.ReadWriteTransaction) error {
+	rs, err := sd.client.ReadWriteTransactionWithOptions(ctx, func(ctx context.Context, spannerRWT *spanner.ReadWriteTransaction) error {
 		txSource := func() readTX {
 			return &traceableRTX{delegate: spannerRWT}
 		}
 
+		if config.Metadata != nil {
+			// Insert the metadata into the transaction metadata table.
+			mutation := spanner.Insert(tableTransactionMetadata,
+				[]string{colTransactionTag, colMetadata},
+				[]any{transactionTag, config.Metadata.AsMap()},
```

Comment:
are you planning to add limit in the size of the metadata at the service level? (i.e. validate this in `WriteRelationships` and `DeleteRelationships`)?

Reply:
It would be done at the API level; that has not been added to the API yet (it will be done in a followup PR)